### PR TITLE
feat: make smartHandler more configurable

### PR DESCRIPTION
**Pull request by rsmayda**
Description of changes:
- Added more documentation
- updated THIRD_PARTY file
- Moved the is 'Practitioner' (Admin) check to the handler and not the helper
- Making adminAccessTypes & bulkDataAccessTypes configurable

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
 ## Code of Conduct
 
-This project has adopted the [Amazon Open Source Code of Conduct](https://aws.github.io/code-of-conduct).
```
Comment (rsmayda) [inline CODE_OF_CONDUCT.md:3]: No content change just cleaned up needless newlines

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,17 +1,14 @@
 # Contributing Guidelines
 
-Thank you for your interest in contributing to our project. Whether it's a bug report, new feature, correction, or additional
```
Comment (rsmayda) [inline CONTRIBUTING.md:3]: No content change just cleaned up needless newlines

File: README.md [hunk 3]
```diff
@@ -2,34 +2,110 @@
...
+- For `user` scopes there must be a `fhirUser` claim in the access_token
+- The access modifiers `read` and `write` will follow the [SMARTConfig](./src/smartConfig.ts) given
+
+### Attribute Based Access Control (ABAC)
+
+This implementation of the SMART on FHIR specification uses attribute based access control. Access to a resource is given if any of the following are true:
+
+- is the fhirUser making the request considered an Admin (default configuration makes a Practitioner an admin)
+- is the fhirUser making the request or the patient in context looking up their own resource (verified via the `resourceType` and `id`)
+- is the fhirUser making the request or the patient in context referenced in the resource they are taking action on
```
Comment (carvantes) [inline README.md]: nit: I think these bullet points read better as:
`The fhirUser making the request is considered an Admin (default configuration makes a Practitioner an admin)`

**Review by carvantes — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -2,34 +2,110 @@
...
+
+- For `patient` scopes there must be a `launch_response_patient` claim in the access_token
+- For `user` scopes there must be a `fhirUser` claim in the access_token
+- The access modifiers `read` and `write` will follow the [SMARTConfig](./src/smartConfig.ts) given
+
+### Attribute Based Access Control (ABAC)
+
+This implementation of the SMART on FHIR specification uses attribute based access control. Access to a resource is given if any of the following are true:
+
+- The fhirUser making the request considered an Admin (default configuration makes a Practitioner an admin)
```
Comment (nguyen102) [inline README.md]: "The fhirUser making the request considered" => "The fhirUser making the request **is** considered"

**Review by nguyen102 — DISMISSED** [review]
Looks good, very thorough documentation. One minor text change suggestion.

One question: Will we be linking users to a reference Auth Server implementation anywhere? Specifically the Okta Auth server?

File: README.md [hunk 5]
```diff
@@ -95,17 +94,17 @@ What we mean by 'referenced in the resource' is the `reference` fields found in
 
 ## Usage
 
-For usage please add this package to your `package.json` file and install as a dependency. For usage examples please see the deployment component's [package.json](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/smart-mainline/package.json)
+Add this package to your `package.json` file and install as a dependency. For usage examples please see the deployment component's [package.json](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/smart-mainline/package.json)
 
 ### Configuration
 
-This SMART implementation is meant to be flexible. It achieves this through the [SMARTConfig](./src/smartConfig.ts). This FHIR server knows the authorization server's jwks endpoint by the configuration passed in.
+The SMART specification gives a lot of room for interpretation between the resource and authorization server relationship. With this in mind we developed our SMART implementation to be flexible. The configurations currently avaibile can be viewed in the [SMARTConfig](./src/smartConfig.ts).
```
Comment (nguyen102) [inline README.md]: avaibile => available

File: README.md [hunk 6]
```diff
@@ -95,17 +94,17 @@ What we mean by 'referenced in the resource' is the `reference` fields found in
...
 
 ### SMART on FHIR scope rules
 
-Within the [SMARTConfig](./src/smartConfig.ts) you can see an example implementation of a ScopeRule. The ScopeRule lays what operations a scope gives access to. For example for the `user/*.write` scope may give 'create' access but not 'update'.
+Within the [SMARTConfig](./src/smartConfig.ts) you can see an example implementation of a ScopeRule. The ScopeRule lays what operations a scope gives access to. For example for the `user/*.write` scope may provide 'create' resource access but not 'update' resource.
```
Comment (nguyen102) [inline README.md]: For example --for-- the `user/*.write` scope --may-- provide 'create' resource access but not 'update' resource.

Remove "for" and "may".

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**